src/main/java/com/tstepnik/planner/controller/TaskController.java

```diff
@@ -0,0 +1,58 @@
+package com.tstepnik.planner.controller;
+
+import com.tstepnik.planner.domain.Task;
+import com.tstepnik.planner.service.TaskService;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.security.Principal;
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/tasks")
+public class TaskController {
+
+    private final TaskService taskService;
+
+    public TaskController(TaskService taskService) {
+        this.taskService = taskService;
+    }
+
+    @PreAuthorize("hasRole('ADMIN')")
+    @GetMapping("/all")
+    public ResponseEntity<List<Task>> tasks() {
+        return ResponseEntity.ok(taskService.getTasks());
+    }
+
+    @PreAuthorize("hasRole('USER')")
+    @GetMapping
```

Comment:
I am not sure about this mapping. I'd rather use `/mine` or use the same endpoint for admin and users to get list of tasks.

Reply:
But it make errors, it have to be different. User has acess only for his task, admin for all tasks.

Reply:
You can make it work with additional logic. I know that having 2 endpoints with same url gives errors. That's why i said "use the same endpoint for admin and users" - same endpoint, not same url

Reply:
But endpoint logic is different, user return only his task, admin get acess to all task, how can i do it? I have been thinking about add if but I don't know how create boolean.